### ISSUE TITLE
PHP Notice:  String offset cast occurred in 

### ISSUE THREAD

**Issue by hex0wn** [deceze/BloomFilter#1]
`PHP Notice:  String offset cast occurred in`, is it OK ?

**Comment by deceze** [comment]
Can you please provide a complete test case and a complete error message? Also information about your PHP version? This doesn't really tell me anything at the moment.

**Comment by hex0wn** [comment]
My PHP `7.03` was compiled from sources on Ubuntu14.04. 

``` php
require 'BloomFilter.php';

$bf = new BloomFilter(40000, 14);
$bf->add('foo');

if ($bf->maybeInSet('foo')) {
        echo 'foo may be in set';
}
```

```
Notice: String offset cast occurred in /var/www/BloomFilter/BloomFilter.php on line 67

Call Stack:
    0.0002     360544   1. {main}() /var/www/BloomFilter/test.php:0
    0.0004     393024   2. BloomFilter->add(string(3)) /var/www/BloomFilter/test.php:5
    0.0006     393488   3. BloomFilter->setBitAtPosition(long) /var/www/BloomFilter/BloomFilter.php:128

```

I run it from command line. Result is right, but what happened?

**Comment by LeMoussel** [comment]
:+1:  I got same error Notice message with PHP `5.6.4`  in this functions

```
protected function setBitAtPosition($pos) {
    list($char, $byte) = $this->position2CharAndByte($pos);
    // Error Notice :  String offset cast occurred in ....
    $this->bitField[$char] = $this->bitField[$char] | $byte;
}

protected function getBitAtPosition($pos) {
    list($char, $byte) = $this->position2CharAndByte($pos);
    // Error Notice :  String offset cast occurred in ....
    return ($this->bitField[$char] & $byte) === $byte;
}
```

This is a new warning that they introduced in 5.4 because it was a frequent mistake and a frequent cause of headaches. As of PHP 5.4 string offsets have to either be integers or integer-like strings, otherwise a warning will be thrown.

**Solution:**
Cast to integer.

```
protected function setBitAtPosition($pos) {
    list($char, $byte) = $this->position2CharAndByte($pos);
    $this->bitField[(int)$char] = $this->bitField[(int)$char] | $byte;
}

protected function getBitAtPosition($pos) {
    list($char, $byte) = $this->position2CharAndByte($pos);
    return ($this->bitField[(int)$char] & $byte) === $byte;
}
```

**Comment by deceze** [comment]
Sorry for the late response, was quite busy recently.
This is fixed now in 76efba6c0014f8643a04c6c72471f5b8e67d0cd5.
